### PR TITLE
Keys generation

### PR DESCRIPTION
**Pull request by swamyb3679**
Keys generation script

### CODE REVIEW

**Comment by DukeDavis12** [comment]
Please sign off your commit. Use `-s` flag while adding commits.

File: component-samples/scripts/keys_gen.sh [hunk 1]
```diff
@@ -0,0 +1,243 @@
+#!/bin/bash
```
Comment (DukeDavis12) [inline component-samples/scripts/keys_gen.sh:1]: License header missing

Comment (swamyb3679) [reply component-samples/scripts/keys_gen.sh:1]: Added the licence header in the suggested file.

File: component-samples/scripts/keys_gen.sh [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+COMP_PATH=$1
+CERTS_PATH=$COMP_PATH/creds
+MFG_CERTS_PATH=$COMP_PATH/${components[0]}
+OWN_CERTS_PATH=$COMP_PATH/${components[1]}
+RES_CERTS_PATH=$COMP_PATH/${components[2]}
+
+usage()
+{
+        echo -e "Usage:
+        $0  <absolute_path_of_pri_components>  <OPTIONS>\n
```
Comment (trbehera) [inline component-samples/scripts/keys_gen.sh:21]: absolute_path_of_pri_components => /path/to/root/of/pri/binaries

File: component-samples/scripts/keys_gen.sh [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+                -d => To create device pem files
+                -h => Prints help message\n
+        NOTE: Order of arguments passing shouldn't change"
+}
+
+keystore_gen()
+{
+        PASS_KEY=`openssl rand --base64 12 | tr -dc 0-9A-Za-z`
+        # RSA2048 Keys Generations
+        openssl genrsa -F4 -out $CERTS_PATH/${1}_rsa2048_private.key 2048
```
Comment (trbehera) [inline component-samples/scripts/keys_gen.sh:36]: Check if we can redirect the majority of log messages to some file and only print information to track status of the outcome.

File: component-samples/scripts/keys_gen.sh [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+        keytool -list -keystore $CERTS_PATH/${1}_keystore.p12 -storepass ${PASS_KEY}
+
+        # Creating a keys.pem files
+        cat $CERTS_PATH/${1}_ec256_public.key | tee $CERTS_PATH/${1}_pub_keys.pem > /dev/null
+        cat $CERTS_PATH/${1}_ec384_public.key | tee -a $CERTS_PATH/${1}_pub_keys.pem > /dev/null
+        cat $CERTS_PATH/${1}_rsa2048_public.key | tee -a $CERTS_PATH/${1}_pub_keys.pem > /dev/null
+
+        echo "@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@" | \
+                tee $CERTS_PATH/${1}_keys.hash > /dev/null
+        echo "${1} key password is '${PASS_KEY}'" | tee -a $CERTS_PATH/${1}_keys.hash > /dev/null
```
Comment (trbehera) [inline component-samples/scripts/keys_gen.sh:76]: Create separate files for password and hashes .. that might be helpful for automation.

File: component-samples/scripts/keys_gen.sh [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+        # Hash generations
+        openssl x509 -in $CERTS_PATH/${1}_rsa2048_cert.pem -pubkey -noout | openssl enc -base64 -d \
+                > $CERTS_PATH/${1}_rsa2048.der
+        export RSA2048_HASH=`cat $CERTS_PATH/${1}_rsa2048.der | openssl dgst -sha256 | awk '/s/{print toupper($2)}'`
+        openssl x509 -in $CERTS_PATH/${1}_ec256_cert.pem -pubkey -noout | openssl enc -base64 -d \
+                > $CERTS_PATH/${1}_ec256.der
+        export EC256_HASH=`cat $CERTS_PATH/${1}_ec256.der | openssl dgst -sha256 | awk '/s/{print toupper($2)}'`
+        openssl x509 -in $CERTS_PATH/${1}_ec384_cert.pem -pubkey -noout | openssl enc -base64 -d \
+                > $CERTS_PATH/${1}_ec384.der
+        export EC384_HASH=`cat $CERTS_PATH/${1}_ec384.der | openssl dgst -sha256 | awk '/s/{print toupper($2)}'`
```
Comment (trbehera) [inline component-samples/scripts/keys_gen.sh:92]: Is this HASH generation compatible with RV allowlist configuration? If we add this HASH value to RV source code, does it successfully pass the specific public key?

File: component-samples/scripts/keys_gen.sh [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+        done
+        for i in $CERTS_PATH/*.hash; do if [[ $i =~ ".hash" ]]; then update_rv_hashes; break; fi; done
+}
+
+start_generation $*
+
+# Final status of the whole execution
+if [[ `ls $CERTS_PATH | wc -l` -gt 0 ]]; then
+        echo "@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@"
+        echo "##################### ${bold}KEYSTORE GENERATION IS SUCCESSFUL${normal} ######################"
```
Comment (trbehera) [inline component-samples/scripts/keys_gen.sh:238]: Remove the prints with @ and # characters unless that is used by automation to filter some statement. Also, remove ${bold} and ${normal} comments as sometimes they create issues while parsing the generated output.

File: component-samples/scripts/keys_gen.sh [hunk 7]
```diff
@@ -0,0 +1,247 @@
+#!/bin/bash
+#
+# Copyright 2021 Intel Corporation
+# SPDX-License-Identifier: Apache 2.0
+
```
Comment (trbehera) [inline component-samples/scripts/keys_gen.sh:5]: Add comments about what this script does and also explains the naming of different files.

That way, we don't have to add metadata information in the password / hash files.

File: component-samples/scripts/keys_gen.sh (hunk 2 above)
Comment (swamyb3679) [reply component-samples/scripts/keys_gen.sh:46]: Updated

File: component-samples/scripts/keys_gen.sh (hunk 3 above)
Comment (swamyb3679) [reply component-samples/scripts/keys_gen.sh]: Updated

File: component-samples/scripts/keys_gen.sh (hunk 4 above)
Comment (swamyb3679) [reply component-samples/scripts/keys_gen.sh]: Updated

File: component-samples/scripts/keys_gen.sh (hunk 5 above)
Comment (swamyb3679) [reply component-samples/scripts/keys_gen.sh:134]: Validated this scenario earlier. Scenario works fine.

File: component-samples/scripts/keys_gen.sh (hunk 6 above)
Comment (swamyb3679) [reply component-samples/scripts/keys_gen.sh]: Removed the prints with @, #, ${bold} and ${normal}.

File: component-samples/scripts/keys_gen.sh (hunk 7 above)
Comment (swamyb3679) [reply component-samples/scripts/keys_gen.sh:30]: Added all the comments.

**Review by trbehera — APPROVED** [review]

**Review by DukeDavis12 — APPROVED** [review]

**Pull request merged**